onboarding/src/Components/QnA/QnA.js

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState, useRef } from "react";
+import { v4 as uuidv4 } from "uuid";
+import MarkdownArea from "../MarkdownArea";
+import { deleteQnA } from "../hooks/QnAAPI";
+
+function QnA({
+  id,
+  question,
+  answer,
+  order,
+  qaList,
+  setQaList,
+  maxOrder,
+  setMaxOrder,
+  draggableProps,
+  innerRef,
+  dragHandleProps,
+  editMode,
+}) {
+  const [q, setQuestion] = useState(question);
+  const [a, setAnswer] = useState(answer);
+  const refValue = useRef(qaList);
+
+  useEffect(() => {
+    refValue.current = qaList;
+  });
+
+  const changeQuestion = (content) => {
+    setQuestion(content);
+    const updatedList = refValue.current.map((qa) => {
+      if (qa.id === id) {
+        qa.question = content;
+      }
+      return qa;
+    });
+    setQaList(updatedList);
+  };
+
+  const changeAnswer = (content) => {
+    setAnswer(content);
+    const updatedList = refValue.current.map((qa) => {
+      if (qa.id === id) {
+        qa.answer = content;
+      }
+      return qa;
+    });
+    setQaList(updatedList);
+  };
+
+  const handleCopyQA = (e) => {
+    e.preventDefault();
+    const updatedList = refValue.current.map((qa) => {
+      qa.order > order ? (qa.order = qa.order + 1) : qa;
+      return qa;
+    });
+    const copiedQnA = {
+      question: question,
+      answer: answer,
+      order: order + 1,
+      id: uuidv4(),
+    };
+    const index = qaList.findIndex((qa) => qa.id === id);
+
+    updatedList.splice(index + 1, 0, copiedQnA);
+    setQaList(updatedList);
+    setMaxOrder(maxOrder + 1);
+  };
+
+  const handleDeleteQnA = (e) => {
+    e.preventDefault();
+    const listWithoutDeletedItem = qaList.filter((item) => item.id !== id);
+    const updatedList = listWithoutDeletedItem.map((qa, index) => {
+      qa.order = index + 1;
+      return qa;
+    });
+
+    setMaxOrder(maxOrder - 1);
+    setQaList(updatedList);
+    if (typeof id === "number") deleteQnA(id);
+  };
+
+  const preview = (
+    <div className="task-issue">
+      <div className="card">
+        <div className="card-body">
+          <div className="form-group">
+            <p
+              className="m-0"
+              dangerouslySetInnerHTML={{ __html: question }}
+            ></p>
+          </div>
+          <hr />
+          <p className="m-0" dangerouslySetInnerHTML={{ __html: answer }}></p>
```

Comment:
Scary thing of concern.
What's wrong with
`<p>{ answer }</p>` or using state?
I suspect it's something with `htmlspecialchars` (encode/decode).
I believe there is solution for this somewhere.
But it passed common test for `XSS`
`<script>alert("XSS')</script>`
`%3cscript%3ealert("XSS")%3c%2fscript%3e`
`\u003cscript\u003ealert("XSS")\u003c\u002fscript\u003e`